recipes/osx-sysroot/activate-mdt.sh

```diff
@@ -0,0 +1,65 @@
+LOCAL_MACOSX_DEPLOYMENT_TARGET="@MACOSX_DEPLOYMENT_TARGET@"
+
+if [ -z "${MACOSX_SDK_VERSION}" ] && [ $(uname) = "Darwin" ]; then
+    MACOSX_SDK_VERSION=$(xcrun --sdk macosx --show-sdk-version)
+fi
+
+if [ ! -z "$MACOSX_SDK_VERSION" ]; then
+    # read version into array (default syntax uses space separation, so just replace the dots)
+    # this works across zsh, sh and bash
+    sdk_parts=($(sh -c "var=$MACOSX_SDK_VERSION; echo \${var//'.'/' '}"))
+    mdt_parts=($(sh -c "var=$LOCAL_MACOSX_DEPLOYMENT_TARGET; echo \${var//'.'/' '}"))
+
+    # zsh starts indexes at 1 and bash at 0
+    # however both languages return an empty string for indexes out of range
+    # so we loop over all of 0 1 2 3 and compare missing things as equal
+    error_out=0
+    for (( i=0; i<=3; ++i )); do
+        if [ "${sdk_parts[$i]}" = "" ]; then
+            sdk_part=0
+        else
+            sdk_part=${sdk_parts[$i]}
+        fi
+
+        if [ "${mdt_parts[$i]}" = "" ]; then
+            mdt_part=0
+        else
+            mdt_part=${mdt_parts[$i]}
+        fi
+
+        if [ $sdk_part -lt $mdt_part ]; then
+            error_out=1
+            break
+        elif [ $sdk_part -gt $mdt_part ]; then
+            break
+        fi
+    done
+
+    if [ $error_out -eq 1 ]; then
+        echo "ERROR: MACOSX_DEPLOYMENT_TARGET (${LOCAL_MACOSX_DEPLOYMENT_TARGET}) \
+must be less than or equal to the MacOS SDK version (${MACOSX_SDK_VERSION}). Set MACOSX_SDK_VERSION in the \
+conda_build_config.yaml file of your recipe to a version greater than or equal to the MACOSX_DEPLOYMENT_TARGET."
```

Comment:
This is not an issue. If the SDK is say 10.13 and deployment target is 11, you would get a binary that can run on `>=10.13` since it won't use any newer features.

Reply:
oh really? blarg ok

Reply:
I mean, that still sounds like an issue worth warning about? If someone specifies deployment target 11, and then gets a compilation error (likely) with an older SDK that we could have prevented with a clear instruction, why not do that?

Reply:
Isuru is saying, I think, there will never be a compilation error?

Reply:
I doubt that compilation can succeed when a given project needs newer features (otherwise why raise the deployment target in the first place?), but if people want to avoid this (IMO beneficial) warning, then that's not a hill worth dying on for me.